core/load_balancer_rule_test.go

```diff
@@ -306,6 +306,86 @@ func TestLoadBalancerRulesClient_Get(t *testing.T) {
 	}
 }
 
+func TestLoadBalancerRulesClient_Get2(t *testing.T) {
+	type args struct {
+		ref LoadBalancerRuleRef
+	}
+	tests := []struct {
+		name    string
+		frm     fakeRequestMakerArgs
+		args    args
+		resp    *katapult.Response
+		respErr error
+		respV   *loadBalancerRulesResponseBody
+		want    *LoadBalancerRule
+		wantReq *katapult.Request
+		wantErr string
+	}{
+		{
+			name: "success",
+			args: args{
+				ref: LoadBalancerRuleRef{ID: "123"},
+			},
+			respV: &loadBalancerRulesResponseBody{
+				LoadBalancerRule: &LoadBalancerRule{
+					ID:         "123",
+					ListenPort: 132,
+				},
+			},
+			want: &LoadBalancerRule{
+				ID:         "123",
+				ListenPort: 132,
+			},
+			wantReq: &katapult.Request{
+				Method: "GET",
+				URL: &url.URL{
+					Path: "/core/v1/load_balancers/rules/_",
+					RawQuery: url.Values{
+						"load_balancer_rule[id]": []string{"123"},
+					}.Encode(),
+				},
+			},
+		},
+		{
+			name: "request error",
+			args: args{
+				ref: LoadBalancerRuleRef{ID: "123"},
+			},
+			respErr: fmt.Errorf("flux capacitor undercharged"),
+			wantErr: "flux capacitor undercharged",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			frm := newFakeRequestMaker2(t, tt.resp, tt.respErr, tt.respV)
+
+			c := NewLoadBalancerRulesClient(frm)
+
+			got, resp, err := c.Get(
+				context.Background(),
+				tt.args.ref,
+			)
+
+			assert.Equal(t, 1, len(frm.Calls), "only 1 request should be made")
```

Comment:
I wonder if there's added value to use taking these common asserts and adding a frm.AssertExpectations(t) method ?

Reply:
Potentially, but that might be better served as just an assertion helper function that's otherwise not attached to the frm. And I kind of prefer having the test body be clearly explicit about what is supposed to happen, rather than having a bunch of logic hidden away in a generically named function. So I'd at least vote for having descriptively named assertion helper functions.